### PR TITLE
Empty intervals needs to start in position -1

### LINKED ISSUE #89789: Sub-iterators of ConjunctionDISI are not on the same document! error
### Elasticsearch Version

8.4.1, 7.17.5

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

official docker image on Darwin MacBookBertrand.local 21.5.0 Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:22 PDT 2022; root:xnu-8020.121.3~4/RELEASE_X86_64 x86_64

### Problem Description

Using an index with mapping containing a nested field, we get the "Sub-iterators of ConjunctionDISI are not on the same document!" error if the search use an intervals query and the searched value is empty (after stop words filtering for example).

### Steps to Reproduce

1. Create a simple index with a nested field and stop words filter
```
PUT /simple_test
{
    "settings": {
        "analysis": {
            "analyzer": {
                "default": {
                    "tokenizer": "whitespace",
                    "filter": [
                        "stop"
                    ]
                }
            }
        }
    },
    "mappings": {
        "properties": {
            "object": {
                "type": "text"
            },
            "documents": {
                "type": "nested",
                "properties": {
                    "content": {
                        "type": "text"
                    }
                }
            }
        }
    }
}
```

2. Add a document
```
POST /simple_test/_doc/1
{
    "object": "test"
}
```
No matter if the nested field has content or not

3. Execute a query with stop words
```
GET /simple_test/_search
{
    "query": {
        "intervals": {
            "object": {
                "match": {
                    "query": "not in that",
                    "max_gaps": 2,
                    "ordered": false
                }
            }
        }
    }
}
```

As the searched words are all filtered, the search result is
```
{
    "error": {
        "root_cause": [
            {
                "type": "illegal_argument_exception",
                "reason": "Sub-iterators of ConjunctionDISI are not on the same document!"
            }
        ],
        "type": "search_phase_execution_exception",
        "reason": "all shards failed",
        "phase": "query",
        "grouped": true,
        "failed_shards": [
            {
                "shard": 0,
                "index": "simple_test",
                "node": "gstAliy6ReeeL5eSvg1IZQ",
                "reason": {
                    "type": "illegal_argument_exception",
                    "reason": "Sub-iterators of ConjunctionDISI are not on the same document!"
                }
            }
        ],
        "caused_by": {
            "type": "illegal_argument_exception",
            "reason": "Sub-iterators of ConjunctionDISI are not on the same document!",
            "caused_by": {
                "type": "illegal_argument_exception",
                "reason": "Sub-iterators of ConjunctionDISI are not on the same document!"
            }
        }
    },
    "status": 400
}
```

### Logs (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by romseygeek**
If a match interval query ends up with no tokens after analysis, we
return an IntervalsSource that produces an empty interval iterator.  This
empty iterator previously always reported its current doc id as 
`NO_MORE_DOCS`.  However, unpositioned DocIdSetIterators should 
report their doc id as -1, so this broke the API contract, which could lead
to exceptions when an empty interval query was combined in a conjunction.
This commit fixes the empty interval implementation so that it returns -1
when unpositioned.

Fixes #89789

### CODE REVIEW

**Comment by romseygeek** [comment]
This bug also appears in the lucene sources.  I'll open a separate fix there.  We should also probably stop maintaining our own IntervalBuilder code here and use the lucene version directly.

File: server/src/internalClusterTest/java/org/elasticsearch/search/query/NestedSearchIT.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+        SearchResponse resp = client().prepareSearch("nested")
+            .setQuery(
+                new IntervalQueryBuilder("empty_text", new IntervalsSourceProvider.Match("an empty query", 0, true, null, null, null))
+            )
+            .get();
+        assertEquals(0, resp.getFailedShards());
+    }
+
+    private static class EmptyAnalyzer extends Analyzer {
```
Comment (javanna) [inline server/src/internalClusterTest/java/org/elasticsearch/search/query/NestedSearchIT.java:67]: is the empty analyzer what requires a full blown IT test? Can't we use a custom analyzer from a unit test?

File: server/src/internalClusterTest/java/org/elasticsearch/search/query/NestedSearchIT.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+
+import static java.util.Collections.singletonMap;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+
+public class NestedSearchIT extends ESIntegTestCase {
```
Comment (javanna) [inline server/src/internalClusterTest/java/org/elasticsearch/search/query/NestedSearchIT.java:34]: With the plugin that it requires, this is very specific test for the scenario that we want to fix. Could we rename the class to something more specific?

**Review by javanna — APPROVED** [review]
left two nits, LGTM otherwise

File: server/src/internalClusterTest/java/org/elasticsearch/search/query/NestedSearchIT.java (hunk 1 above)
Comment (romseygeek) [reply server/src/internalClusterTest/java/org/elasticsearch/search/query/NestedSearchIT.java:67]: We need the full blown IT test to create the index and run searches over it

File: server/src/internalClusterTest/java/org/elasticsearch/search/query/NestedSearchIT.java (hunk 2 above)
Comment (romseygeek) [reply server/src/internalClusterTest/java/org/elasticsearch/search/query/NestedSearchIT.java:34]: How about IntervalQueriesIT? I can see this being a useful place to add further tests in future.

Comment (javanna) [reply server/src/internalClusterTest/java/org/elasticsearch/search/query/NestedSearchIT.java:34]: Sounds good to me, I guess that other interval things can be added later despite that weird plugin is loaded?

File: server/src/internalClusterTest/java/org/elasticsearch/search/query/NestedSearchIT.java (hunk 1 above)
Comment (javanna) [reply server/src/internalClusterTest/java/org/elasticsearch/search/query/NestedSearchIT.java:67]: yep I figured.

**Pull request merged**